ui/lib/sync/addon/routes/secrets/destinations/destination.ts

```diff
@@ -5,37 +5,60 @@
 
 import Route from '@ember/routing/route';
 import { service } from '@ember/service';
+import apiMethodResolver from 'sync/utils/api-method-resolver';
 
-import type Store from '@ember-data/store';
 import type RouterService from '@ember/routing/router-service';
 import type FlashMessageService from 'vault/services/flash-messages';
 import type Transition from '@ember/routing/transition';
-import type SyncDestinationModel from 'vault/models/sync/destination';
+import type ApiService from 'vault/services/api';
+import type { Destination } from 'vault/sync';
+import type CapabilitiesService from 'vault/services/capabilities';
+import type { CapabilitiesMap } from 'vault/app-types';
 
-interface RouteParams {
-  name?: string;
-  type?: string;
-}
+type Params = {
+  name: string;
+  type: Destination['type'];
+};
+
+export type DestinationRouteModel = {
+  destination: Destination;
+  capabilities: CapabilitiesMap;
+};
 
 export default class SyncSecretsDestinationsDestinationRoute extends Route {
-  @service declare readonly store: Store;
+  @service declare readonly api: ApiService;
   @service('app-router') declare readonly router: RouterService;
   @service declare readonly flashMessages: FlashMessageService;
+  @service declare readonly capabilities: CapabilitiesService;
+
+  async model(params: Params) {
+    const { name, type } = params;
+    const method = apiMethodResolver('read', type);
+
+    const requests = [
+      this.api.sys[method](name, {}),
+      this.capabilities.fetch([
+        this.capabilities.pathFor('syncDestination', { name, type }),
+        this.capabilities.pathFor('syncSetAssociation', { name, type }),
+      ]),
+    ];
+    const [destination, capabilities] = await Promise.all(requests);
```

Comment:
out of curiosity is using `Promise.all()` "in" and `RSVP.hash` "out" of style now?

Reply:
I'm not sure if functionally this is much difference other than one accepts an array of promises and the other an object. I like not having to import `hash` I guess.

Reply:
Yep, agreed. Was mostly wondering if there was some new trend I missed out on